boot/zephyr/main.c

```diff
@@ -146,9 +140,18 @@ static void do_boot(struct boot_rsp *rsp)
     cleanup_arm_nvic(); /* cleanup NVIC registers */
 #endif
 
-#ifdef CONFIG_BOOT_INTR_VEC_RELOC
-    VTOR = vt;
+#if defined(CONFIG_SW_VECTOR_RELAY) && defined(CONFIG_BOOT_INTR_VEC_RELOC)
```

Comment:
Maybe this could be some more legible, as  CONFIG_BOOT_INTR_VEC_RELOC is common here:
```suggestion
#ifdef CONFIG_BOOT_INTR_VEC_RELOC
#if defined(CONFIG_SW_VECTOR_RELAY)
    _vector_table_pointer = vt;
#if defined(CONFIG_CPU_CORTEX_M_HAS_VTOR)
    /* Set VTOR to point to the relay vector table before the chain-load. */
    SCB->VTOR = (uint32_t)__vector_relay_table;
#endif
#elif  defined(CONFIG_CPU_CORTEX_M_HAS_VTOR)
    SCB->VTOR = vt;
#endif /* CONFIG_BOOT_INTR_VEC_RELOC */
```

Reply:
Done.